tests/unittests/config/test_cc_ubuntu_advantage.py

```diff
@@ -197,160 +210,218 @@ def test_schema_validation(self, config, error_msg):
                 validate_cloudconfig_schema(config, get_schema(), strict=True)
 
 
-class TestHandle(CiTestCase):
-
-    with_logs = True
-
-    def setUp(self):
-        super(TestHandle, self).setUp()
-        self.tmp = self.tmp_dir()
+class TestHandle:
 
-    @mock.patch("%s.maybe_install_ua_tools" % MPATH)
-    def test_handle_no_config(self, m_maybe_install_ua_tools):
-        """When no ua-related configuration is provided, nothing happens."""
-        cfg = {}
-        handle("ua-test", cfg=cfg, cloud=None, log=self.logger, args=None)
-        self.assertIn(
-            "DEBUG: Skipping module named ua-test, no 'ubuntu_advantage'"
-            " configuration found",
-            self.logs.getvalue(),
-        )
-        self.assertEqual(m_maybe_install_ua_tools.call_count, 0)
-
-    @mock.patch("%s.configure_ua" % MPATH)
-    @mock.patch("%s.maybe_install_ua_tools" % MPATH)
-    def test_handle_tries_to_install_ubuntu_advantage_tools(
-        self, m_install, m_cfg
-    ):
-        """If ubuntu_advantage is provided, try installing ua-tools package."""
-        cfg = {"ubuntu_advantage": {"token": "valid"}}
-        mycloud = FakeCloud(None)
-        handle("nomatter", cfg=cfg, cloud=mycloud, log=self.logger, args=None)
-        m_install.assert_called_once_with(mycloud)
+    mycloud = FakeCloud(None)
 
-    @mock.patch("%s.configure_ua" % MPATH)
-    @mock.patch("%s.maybe_install_ua_tools" % MPATH)
-    def test_handle_passes_credentials_and_services_to_configure_ua(
-        self, m_install, m_configure_ua
+    @pytest.mark.parametrize(
+        [
+            "cfg",
+            "handle_kwargs",
```

Comment:
From the looks of params provided, this could just be `cloud` and you can then just invoke the following inline in the test:
```python
handle("nomatter", cfg=cfg, cloud=cloud, log=None, args=None)
```


only preference here is really just that the extra repeated dict in each param is just more stuff to look at and ignore.

Reply:
Much simpler, thanks.